bin/pre-commit-hook.js

```diff
@@ -28,3 +32,28 @@ if ( lintResult.status ) {
 		'repeat the commit command with --no-verify to avoid this check.' );
 	process.exit( 1 );
 }
+
+
+// run prettier for any files in the commit that have @format within their first docblock
+const formatRegex = /@format/;
+const firstDocBlockStart = /\/\*\*/;
+const firstDocBlockEnd = /\*\//;
+
+files
+	.map( file => path.join( __dirname, '../', file ) )
+	.forEach( file => {
+		fs.readFile( file, 'utf8', ( err, data ) => {
+			const firstDocBlockStartIndex = _.get( firstDocBlockStart.exec( data ), 'index' );
+			const firstDocBlockEndIndex = _.get( firstDocBlockEnd.exec( data ), 'index' );
+			const firstDocBlockText =
+				_.isInteger( firstDocBlockStartIndex ) &&
+				_.isInteger( firstDocBlockEndIndex ) &&
+				data.substring( firstDocBlockStartIndex, firstDocBlockEndIndex + 2 );
+
+			if ( firstDocBlockText && formatRegex.test( firstDocBlockText ) ) {
+				const formattedData = prettier.format( data, {} );
+				fs.writeFileSync( file, formattedData );
+				execSync( `git add ${ file }` );
+			}
```

Comment:
> I'd still argue that the regex-only method would be harder to maintain/understand than the verbose-but-simple logic

I agree, though yours is (to no offense) _more wrong_ than mine because it won't allow for normal block comments before the docblock (shouldn't be an issue) _and_ slower :wink:

there's something odd about the arrangement of the logic here that stuck out to me. did you consider making it somewhat more monadic in structure - the so-called train-tracks of early abort?

```js
const hasIndicator( s ) {
	const start = s.indexOf( '/**' ); // find first docblock opener
	if ( -1 === start ) {
		return false;
	}

	const end = s.indexOf( '*/', start ); // find first comment closing inside docblock
	if ( -1 === end ) {
		return false;
	}

	return /@format pretty/.test( s.slice( start, end ) );
}
```